### ISSUE TITLE
[no-var-requires] false negative when require statement includes a type cast

### ISSUE THREAD

**Issue by adidahiya** [typescript-eslint/typescript-eslint#2125]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-var-requires": "error",
  }
}
```

```TS
const myJson = require("./someFile.json") as MyJson;
```

(initially encountered in [this PR](https://github.com/palantir/blueprint/pull/4158/files#diff-aa0e7de5c6a62de9407b7dd00098dd2aR27))

**Expected Result**

lint error on this require statement, just like the TSLint rule

**Actual Result**

no error

**Additional Info**

easy enough to work around, since I can change the syntax and the rule will report the error as expected:

```ts
const myJson: MyJson = require("./someFile.json");
```

.. but nonetheless I think the rule should report an error in both cases.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.0.1` |
| `@typescript-eslint/parser`        | `3.0.1` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.16.2` |

**Comment by adidahiya** [comment]
This also seems to produce a false negative when there is anything chained onto the `require()` statement, like:

```
const mySvg = require("raw-loader!./file.svg").default;
```

... which also diverges from the TSLint rule
